rrs/tools/upgrade_history_internal.py

```diff
@@ -0,0 +1,617 @@
+                                        layersubdir_start)
+
+            if not diffitem.b_path or diffitem.deleted_file or not diffitem.b_path.startswith(layersubdir_start):
+                # Deleted, or moved out of the layer (which we treat as a delete)
+                if typename == 'recipe':
+                    deleted.append(diffitem.a_path)
+                continue
+
+            if typename == 'recipe':
+                (to_typename, _, _) = recipeparse.detect_file_type(diffitem.b_path,
+                                            layersubdir_start)
+                if to_typename == 'recipe':
+                    ct_files.append(os.path.join(repodir, diffitem.b_path))
+                    if diffitem.a_path is None or diffitem.new_file:
+                        added_files.append(diffitem.b_path)
+                if diffitem.a_path != diffitem.b_path:
+                    moved_files.append((diffitem.a_path, diffitem.b_path))
+            elif typename == 'incfile':
+                fpath = os.path.dirname(os.path.join(repodir, diffitem.a_path))
+                if not fpath in incdirs:
+                    incdirs.append(fpath)
+
+    for fpath in incdirs:
+        # Let's just assume that all .bb files next to a .inc need to be checked
+        for f in glob.glob(os.path.join(fpath, '*.bb')):
+            if not f in ct_files:
+                ct_files.append(f)
+
+    # Check moves for recipe -> inc with an added recipe
+    # (i.e. the move should really be to the newly added recipe)
+    # example: d5a95dc8985a42bb7e50bc4e7dc6b012d711ff08 in OE-Core
+    for i,(a,b) in enumerate(moved_files):
+        if b.endswith('.inc'):
+            for af in added_files:
+                # This is naive, but good enough
+                if af.rsplit('_')[0] == a.rsplit('_')[0]:
+                    moved_files[i] = (a,af)
+
+    return ct_files, deleted, moved_files
+
+
+def checkout_layer_deps(layerbranch, commit, fetchdir, logger):
+    """ Check out the repositories for a layer and its dependencies """
+
+    oecore_map = {}
+    if layerbranch.layer.name != 'openembedded-core':
+        # Filter out some bad commits for OE-Core
+        for good_rev, bad_revs in oecore_bad_revs.items():
+            for bad_rev in bad_revs:
+                oecore_map[bad_rev] = good_rev
+
+    # Some layers will be in the same repository, so we only want to check those out once
+    done_repos = []
+    def checkout_layer(lb, lcommit=None, lcommitdate=None, force=False):
+        urldir = str(lb.layer.get_fetch_dir())
+        repodir = os.path.join(fetchdir, urldir)
+        if not repodir in done_repos:
+            if not lcommit:
+                lcommit = utils.runcmd(['git', 'rev-list', '-1', '--before=%s' % lcommitdate, 'origin/master'], repodir, logger=logger).strip()
+            if lb.layer.name == 'openembedded-core':
+                lmapcommit = oecore_map.get(lcommit, None)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are using the `get()` method to retrieve the value for a key from a dictionary. If the key is not present, then `get()` returns the value that is passed in its optional second parameter. If if the key is not present, then `None` is the default return value. Therefore, there is no need to explicitly specify `None` in the second parameter.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)